datajunction-server/datajunction_server/api/sql.py

```diff
@@ -378,6 +380,20 @@ async def get_sql_for_metrics(  # pylint: disable=too-many-locals
         base_verb=access.ResourceRequestVerb.READ,
     )
 
+    # make sure all metrics exist and have correct node type
+    nodes = [
+        await Node.get_by_name(session, node, raise_if_not_exists=True)
```

Comment:
There's a `Node.get_by_names` method that lets you pass in a list and queries for them all at once right? Then you can just compare the counts.

Reply:
I didn't know... yes I could probably use it here.